src/main/kotlin/io/provenance/client/grpc/PbClient.kt

```diff
@@ -135,37 +148,81 @@ class PbClient(
             baseReq.signers[index].signer.sign(signDocBytes).let { ByteString.copyFrom(it) }
         }.let {
             TxOuterClass.TxRaw.newBuilder()
-                .setAuthInfoBytes(authInfoBytes)
-                .setBodyBytes(txBodyBytes)
-                .addAllSignatures(it)
-                .build()
+                    .setAuthInfoBytes(authInfoBytes)
+                    .setBodyBytes(txBodyBytes)
+                    .addAllSignatures(it)
+                    .build()
         }
 
         return cosmosService.broadcastTx(
-            ServiceOuterClass.BroadcastTxRequest.newBuilder().setTxBytes(txRaw.toByteString()).setMode(mode).build()
+                ServiceOuterClass.BroadcastTxRequest.newBuilder().setTxBytes(txRaw.toByteString()).setMode(mode).build()
         )
     }
 
     fun estimateAndBroadcastTx(
-        txBody: TxBody,
-        signers: List<BaseReqSigner>,
-        mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC,
-        gasAdjustment: Double? = null,
+            txBody: TxBody,
+            signers: List<BaseReqSigner>,
+            mode: ServiceOuterClass.BroadcastMode = ServiceOuterClass.BroadcastMode.BROADCAST_MODE_SYNC,
+            gasAdjustment: Double? = null,
     ): ServiceOuterClass.BroadcastTxResponse = baseRequest(
-        txBody = txBody,
-        signers = signers,
-        gasAdjustment = gasAdjustment
+            txBody = txBody,
+            signers = signers,
+            gasAdjustment = gasAdjustment
     ).let { baseReq -> broadcastTx(baseReq, estimateTx(baseReq), mode) }
 
     fun getBaseAccount(bech32Address: String): Auth.BaseAccount =
-        authClient.account(
-            QueryOuterClass.QueryAccountRequest.newBuilder().setAddress(bech32Address).build()
-        ).account.run {
-            when {
-                this.`is`(Auth.BaseAccount::class.java) -> unpack(Auth.BaseAccount::class.java)
-                else -> throw IllegalArgumentException("Account type not handled:$typeUrl")
+            authClient.account(
+                    QueryOuterClass.QueryAccountRequest.newBuilder().setAddress(bech32Address).build()
+            ).account.run {
+                when {
+                    this.`is`(Auth.BaseAccount::class.java) -> unpack(Auth.BaseAccount::class.java)
+                    else -> throw IllegalArgumentException("Account type not handled:$typeUrl")
+                }
             }
-        }
 
+    fun getAcountBalance(bech32Address: String, denom: String): CoinOuterClass.Coin =
```

Comment:
thought we didnt want a bunch of extra stuff in this lib?

Reply:
@arnabmitra There's a bunch of extension methods I added to the provenance repo that would probably be a good place for this functionality: https://github.com/provenance-io/provenance/tree/main/protoBindings/bindings/kotlin/src/main/kotlin/io/provenance/client/protobuf/extensions

You should check those extension methods, because there could be some duplicate functionality.

Reply:
@afremuth-figure  is looking at these extension methods, i haven't really kept up the clinet libs unfortunately :(

Reply:
@mwoods-figure we're going to move them over in the next release